### PR TITLE
[MHV-59273] Added CantFindYourTeam dropdown and moved EditPreferences…

### PR DESCRIPTION
**Pull request by vsaleem**
## Are you removing, renaming or moving a folder in this PR?
- [ ] No, I'm not changing any folders (skip to Summary and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

If the folder you changed contains a `manifest.json`, search for its `entryName` in the content-build [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) (the `entryName` there will match).

If an entry for this folder exists in content-build and you are:
1. **Deleting a folder**: Delete the application entry in [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) and merge that PR **before** this one
- _Add the link to your merged content-build PR here_

2. **Renaming or moving a folder**: Update the entry in the [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json), but do not merge it until your vets-website changes here are merged. The content-build PR must be merged immediately after your vets-website change is merged in to avoid CI errors with content-build (and Tugboat).

### :warning: TeamSites :warning:
Examples of a TeamSite: https://va.gov/health and https://benefits.va.gov/benefits/. This scenario is also referred to as the "injected" header and footer. You can reach out in the `#sitewide-public-websites` Slack channel for questions.

Did you change site-wide styles, platform utilities or other infrastructure?
- [ ] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

> ## Summary
> 
> #### User Story: As a Secure Messaging User, I want to be able to understand what to do if the group on my contact list does not appear in the To: dropdown list
> 
> - Given: User is starting a new message
> 
> - When:  User uses the dropdown menu in the "To" field and does not see the contact the user is looking for.
> 
> - Then:  The user can click on the link, "If you can't find your team" and get more information and help to navigate the problem
> 
> 
> ## Related issue(s)
> - [[MHV-59273]](https://jira.devops.va.gov/browse/MHV-59273) - Add "If you can't find your team" dropdown to SNM page/Move preferences button
> - Complete/latest requirements located in Figma (See Interim solution July 1st section: https://www.figma.com/design/uddjxccqvUyz5FSg1aD5wi/ZFeat---Triage-Group-Selection-Flows?node-id=912-25860&t=3A3ocbuqmkxjcQXS-4
> 
> 
> ## Testing done
> 
> - [ ] Revised and added Unit tests
> - [ ] Automation tests added
> - [ ] Manual testing completed
> - [ ] 508 testing completed 
> 
> NOT acceptable as appropriate testing
> - _Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution records](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)_
> 
> ## Screenshots
> 
> _Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._
> 
> |         | Before | After |
> | ------- | ------ | ----- |
> | Mobile  |image|imageimageimage|
> 
> 
> ## What areas of the site does it impact?
> 
> *Va.gov - Secure Messaging - Start a new message page*
> 
> ## Acceptance criteria
> #### AC1 Dropdown labeled "If you can't find your team" appears on the Start a new message page
> #### AC2 Content for the dropdown is displayed
> #### AC3 Link To MHV Classic SM preferences is displayed and does not open in a new tab
> #### AC4 Link to MHV Classic SM preferences has the focus set to the "My Secure Messaging Contact list" section
> #### AC5 Preferences button has been moved from the top of the page to the bottom
> #### AC6 UCD validation is complete
> 

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/mhv-secure-messaging/components/ComposeForm/CantFindYourTeam.jsx
Comment (oleksii-morgun) [inline src/applications/mhv-secure-messaging/components/ComposeForm/CantFindYourTeam.jsx:16]: I think `useMemo` would be more appropriate and efficient here
```suggestion
  const prefLink = useMemo(
    () => mhvUrl(isAuthenticatedWithSSOe(fullState), 'preferences'),
    [fullState],
  );
```

File: src/applications/mhv-secure-messaging/components/ComposeForm/CantFindYourTeam.jsx
Comment (oleksii-morgun) [inline src/applications/mhv-secure-messaging/components/ComposeForm/CantFindYourTeam.jsx:23]: if these are `false` do we still need to pass them? 
```suggestion

```

File: src/applications/mhv-secure-messaging/components/ComposeForm/CantFindYourTeam.jsx
Comment (oleksii-morgun) [inline src/applications/mhv-secure-messaging/components/ComposeForm/CantFindYourTeam.jsx:52]: I am wondering if this should open in a new tab? for the same reason we discussed during the planning for a link above - not forcing a user to lose progress. Probably worth checking with UCD

File: src/applications/mhv-secure-messaging/components/ComposeForm/RecipientsSelect.jsx
Comment (oleksii-morgun) [inline src/applications/mhv-secure-messaging/components/ComposeForm/RecipientsSelect.jsx:86]: I think this `<CanFindYourTeam />` component should be passed as a child to `<VaSelect>` and we can keep `label` attribute. Doing it the way you do here now, introduces an interesting behavior on error state, which I think we may want to avoid

Comment (oleksii-morgun) [reply src/applications/mhv-secure-messaging/components/ComposeForm/RecipientsSelect.jsx:86]: And you wont need to introduce that additional css styling for `Required` label

**Review by alex-morgun-1 — CHANGES_REQUESTED** [review]
Awesome work @vsaleem Please consider some of these suggestions. Let me know if you may need more clarity on my thoughts here, happy to walk you through.

File: src/applications/mhv-secure-messaging/components/ComposeForm/CantFindYourTeam.jsx
Comment (vsaleem) [reply src/applications/mhv-secure-messaging/components/ComposeForm/CantFindYourTeam.jsx:16]: this is solid 🚀

File: src/applications/mhv-secure-messaging/components/ComposeForm/CantFindYourTeam.jsx
Comment (vsaleem) [reply src/applications/mhv-secure-messaging/components/ComposeForm/CantFindYourTeam.jsx:52]: @oleksii-morgun I confirmed with UCD, their recommendation is to allow this link to open in the same tab for consistency purposes because that’s how it works for the blocked triage group banners. Also, when the contact list lives on [VA.gov](http://va.gov/) we are not opening that in a new tab either.

Comment (oleksii-morgun) [reply src/applications/mhv-secure-messaging/components/ComposeForm/CantFindYourTeam.jsx:52]: in that case, are we properly handling unsaved progress? that is my biggest concern

File: src/applications/mhv-secure-messaging/tests/components/Compose/RecipientsSelect.unit.spec.jsx
Comment (oleksii-morgun) [inline src/applications/mhv-secure-messaging/tests/components/Compose/RecipientsSelect.unit.spec.jsx:106]: no biggie, but this `const` does not need to be capitalized

File: src/applications/mhv-secure-messaging/tests/components/Compose/RecipientsSelect.unit.spec.jsx
Comment (oleksii-morgun) [inline src/applications/mhv-secure-messaging/tests/components/Compose/RecipientsSelect.unit.spec.jsx:112]: this validation may need to be more specific. Are we expecting for a preference deeplink to be rendered? This `querySelector` finds 2 anchor tags, including `find-locations`

File: src/applications/mhv-secure-messaging/tests/components/Compose/RecipientsSelect.unit.spec.jsx
Comment (oleksii-morgun) [inline src/applications/mhv-secure-messaging/tests/components/Compose/RecipientsSelect.unit.spec.jsx:120]: this second expect is probably redundant since you are already including `href` in your `querySelector`. If it finds it, there is no way it can be `.equal('')`

**Review by alex-morgun-1 — DISMISSED** [review]
Approved ✅ pending review by other engineers. @KolbyVA @KyleCardwell please review when you have a moment

**Review by KolbyVA — DISMISSED** [review]

**Review by alex-morgun-1 — APPROVED** [review]

**Pull request merged**
